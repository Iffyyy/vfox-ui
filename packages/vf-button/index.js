import VfButton from './vf-button.vue';
import VfButtonGroup from './vf-button-group.vue';

VfButton.install = function (Vue) {
  Vue.component(VfButton.name, VfButton);
};

VfButtonGroup.install = function (Vue) {
  Vue.component(VfButtonGroup.name, VfButtonGroup);
};


export {
  VfButton,
  VfButtonGroup
}
