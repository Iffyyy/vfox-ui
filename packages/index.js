import { VfButton, VfButtonGroup } from './vf-button/index';
import { version } from '../package.json';

const components = [
  VfButton,
  VfButtonGroup
]

const install = function (Vue) {
  components.map(component => Vue.use(component));
  // shell_components.map(el => {
  //     Vue.component(el.name, el.component);
  // });
}
if (typeof window !== 'undefined' && window.Vue) {
  install(window.Vue)
}

export {
  VfButton,
  VfButtonGroup
}

export default {
  version,
  install
}