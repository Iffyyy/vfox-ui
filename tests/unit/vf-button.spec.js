import { mount } from '@vue/test-utils'
import { VfButton, VfButtonGroup } from '../../packages'
import {  createLocalVue } from '@vue/test-utils';
import UI from '../../packages/index';

const localVue = createLocalVue()
localVue.use(UI)

/////////////////////////////////VfButton

const factory = (props = {}, data = {}) => {
  return mount(VfButton, {
    propsData: {
      ...props
    },
    data() {
      return {
        ...data
      }
    }
  })
}

describe('VfButton', () => {

  it('测试click触发', async () => {
    const wrapper = factory();
    // 获取对应按钮
    const button = wrapper.find('button');
    // 点击按钮
    button.trigger('click');
    await wrapper.vm.$nextTick();
    // 断言$emit('click')函数被触发
    expect(wrapper.emitted().click).toBeTruthy();
  })
})

/////////////////////////////////VfButtonGroup

const factory1 = (props = {}, data = {}) => {
  return mount({
    render() {
      return (
        <VfButtonGroup attrs={{ ...props }}>
          <VfButton type="cloud"> Button1 </VfButton>
          <VfButton type="cloud"> Button2 </VfButton>
        </VfButtonGroup>
      );
    }
  }, {
    propsData: {
      ...props
    },
    data() {
      return {
        ...data
      }
    }
  })
}

describe('VfButtonGroup', () => {
  // 测试内容：snapshot->概括的测试DOM结构
  it('测试DOM结构', () => {
    const wrapper = factory1();
    expect(wrapper.html()).toMatchSnapshot();
  })

  it('测试size属性', () => {
    const wrapper1 = factory1({
      size: 'small'
    });
    // 断言ant-btn-sm类名存在
    expect(wrapper1.classes()).toContain('ant-btn-group-sm');

    const wrapper2 = factory1({
      size: 'large'
    });
    // 断言ant-btn-lg类名存在
    expect(wrapper2.classes()).toContain('ant-btn-group-lg');
  })
})

