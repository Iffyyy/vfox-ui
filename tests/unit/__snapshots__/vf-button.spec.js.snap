// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VfButtonGroup 测试DOM结构 1`] = `<div class="ant-btn-group vf-button-group"><button type="button" class="ant-btn ant-btn-cloud vf-button"><span>Button1</span></button><button type="button" class="ant-btn ant-btn-cloud vf-button"><span>Button2</span></button></div>`;
